worker/worker.go

```diff
@@ -382,16 +382,26 @@ func (w *worker) withHosts(ctx context.Context, contracts []api.ContractMetadata
 	}
 	done := make(chan struct{})
 	go func() {
+		var wg sync.WaitGroup
 		select {
 		case <-done:
 			for _, h := range hosts {
-				w.pool.unlockContract(h.(*sharedSession))
```

Comment:
Fixes an issue where a contract that can't be unlocked for whatever reason prevents the following contracts from being unlocked as well, resulting in a chain reaction of stuck contracts.